service/qos/service_level_controller.hh

```diff
@@ -220,11 +222,23 @@ private:
     };
 
     future<> set_distributed_service_level(sstring name, service_level_options slo, set_service_level_op_type op_type);
-
 public:
+
+    /**
+     *  Register a subscriber for service level configuration changes
+     *  notifications
+     * @param subscriber - a pointer to the subscriber.
+     *
+     * Note: the caller is responsible to keep the pointed to object alive until performing
+     * a call to service_level_controller::unregister_subscriber()).
+     *
+     * Note 2: the subscription is per shard.
+     */
+    void register_subscriber(qos_configuration_change_subscriber* subscriber);
+    future<> unregister_subscriber(qos_configuration_change_subscriber* subscriber);
+
     static sstring default_service_level_name;
 
```

Comment:
I see you made this public, was this intended?

Reply:
It was intended but maybe my intentions are wrong?
We need a subscriber or someone on his behalf will call this API, so it's either that or friendship of `qos_configuration_change_subscriber`

Reply:
I meant `default_service_level_name`. Is it now part of the API?

Reply:
Wait, it was public before.